flask-server/main.py

```diff
@@ -425,6 +425,78 @@ def fetch_username(user_id):
     key = datastore_client.key('user', int(user_id))
     user = list(query.add_filter('__key__', '=', key).fetch(limit=1))[0]
     return user['username'], 200
+    
+# edit bookmark api
+@app.route('/apis/edit-bookmark', methods=['POST'])
+def edit_bookmark():
+    resp_token = decode_auth_token(request.json['Authorization'])
+
+    if is_valid_instance(resp_token):
```

Comment:
can you also check in this if block to verify that the person has write access to the bookmark? (same for clink)